docs/development/adding-a-package-manager.md

```diff
@@ -67,3 +79,23 @@ This is used when more than one package manager shares settings from a common la
 ### `supportsLockFileMaintenance` (optional)
 
 Set to true if this package manager needs to update lock files in addition to package files.
+
+### `updateArtifacts` (async, optional)
+
+This function triggers:
+
+- after a dependency update (for a package file), or
+- during `lockfileMaintenance`
+
+This function is mostly used to directly (or indirectly) update lock files.
+An example of indirectly updating the lock file: running external binaries.
+
+In case of directly updating dependencies in lock files use `updateLockedDependency` instead.
```

Comment:
This is still a bit confusing to me. It looks like we say:

1. Use `updateArtifacts` to directly/indirectly update lock files
2. Use `updateLockedDependency` to update dependencies in lockfiles

I assume that when we update a lockfile we always update the dependencies? I mean if you're not updating dependencies, why touch the lockfile? 😕

Reply:
Contributors should choose `updateLockedDependency` if they are directly manipulating lock files. 

`updateArtifacts` if they are doing it indirectly e.g. by using an external binary or other files.     
But legacy implementations still directly edit lock files here, but we want to move away from that.

Reply:
Can you put your expanded explanation in the docs? I'll review the text and make improvements if needed. 😉